library/src/com/squareup/timessquare/CalendarPickerView.java

```diff
@@ -248,6 +250,77 @@ public FluentInitializer init(Date minDate, Date maxDate) {
     return init(minDate, maxDate, Locale.getDefault());
   }
 
+  /**
+   * {@code datesCollection} must be non-null and all of it's items' {@link Date#getTime()} must
+   * not return 0. Time of all days will be ignored.
+   * For instance, if you pass in a collection of dates: 11/16/2012, 11/17/2012 and 11/20/2012,
+   * Only these particular dates will be selectable. Meaning that 11/18/2012 and 11/19/2012 will
+   * not be selectable although they will fall into {@code minDate} - {@code maxDate} interval.
+   * {@code minDate} will be set to 11/16/2012 and {@code maxDate} to 11/20/2012.
+   * <p>
+   * This will implicitly set the {@link SelectionMode} to {@link SelectionMode#SINGLE}.  If you
+   * want a different selection mode, use {@link FluentInitializer#inMode(SelectionMode)} on the
+   * {@link FluentInitializer} this method returns.
+   * <p>
+   * The calendar will be constructed using the given locale. This means that all names
+   * (months, days) will be in the language of the locale and the weeks start with the day
+   * specified by the locale.
+   *
+   * @param datesCollection collection of selectable dates.
+   */
+  public FluentInitializer init(Collection<Date> datesCollection, Locale locale) {
+    if (datesCollection.isEmpty()){
```

Comment:
Why forbid this? It means I have to push a conditional into calling code if i'm calculating the collection dynamically just to handle an empty collection.